dropwizard-benchmarks/src/main/java/io/dropwizard/benchmarks/jersey/ConstraintViolationBenchmark.java

```diff
@@ -0,0 +1,83 @@
+package io.dropwizard.benchmarks.jersey;
+
+import com.google.common.collect.Iterators;
+import io.dropwizard.jersey.validation.ConstraintMessage;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Valid;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.executable.ExecutableValidator;
+import javax.ws.rs.HeaderParam;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.commons.lang3.reflect.MethodUtils.getAccessibleMethod;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+public class ConstraintViolationBenchmark {
+
+    private ConstraintViolation<ConstraintViolationBenchmark> paramViolation;
+    private ConstraintViolation<ConstraintViolationBenchmark> objViolation;
+
+    public String paramFunc(@HeaderParam("cheese") @NotEmpty String secretSauce) {
+        return secretSauce;
+    }
+
+    public String objectFunc(@Valid Foo foo) {
+        return foo.toString();
+    }
+
+    public class Foo {
+        @NotEmpty
+        private String bar;
+    }
+
+    @Setup
+    public void prepare() {
+        final Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
+        final ExecutableValidator execValidator = validator.forExecutables();
+
+        Set<ConstraintViolation<ConstraintViolationBenchmark>> paramViolations =
+            execValidator.validateParameters(
+                this,
+                getAccessibleMethod(getClass(), "paramFunc", String.class),
+                new Object[]{""} // the parameter value
+            );
+        paramViolation = Iterators.getOnlyElement(paramViolations.iterator());
```

Comment:
I think it's safe just to call `iterator.next`, because we know that the violation is exist.